docker/adamant_env.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+
+set +e
+
+if ! command -v docker &> /dev/null
+then
+  if command -v podman &> /dev/null
+  then
+    function docker() {
+      podman $@
+    }
+  else
+    echo "Neither docker nor podman found!!!"
+    exit 1
+  fi
+fi
+
+DOCKER_CONTAINER_NAME="adamant_container"
+DOCKER_IMAGE_NAME="ghcr.io/lasp/adamant:latest"
+DOCKER_COMPOSE_COMMAND="docker compose"
+export DOCKER_CONTAINER_NAME
+export DOCKER_IMAGE_NAME
+export DOCKER_COMPOSE_COMMAND
+${DOCKER_COMPOSE_COMMAND} version &> /dev/null
+if [ "$?" -ne 0 ]; then
+  export DOCKER_COMPOSE_COMMAND="docker-compose"
+fi
+
+# Helper function to print out command as executed:
+execute () {
+  echo "$ $@"
+  eval "$@"
+}
+
+set -e
+
+usage() {
+  echo "Usage: $1 [start, stop, login, push, build, remove]" >&2
+  echo "*  start: create and start the adamant container" >&2
+  echo "*  stop: stop the running adamant container" >&2
+  echo "*  login: login to the adamant container" >&2
+  echo "*  push: push the image to the Docker registry" >&2
+  echo "*  build: build the image from the Dockerfile" >&2
+  echo "*  remove: remove network and volumes for adamant" >&2
+  exit 1
+}
+
+if [ "$#" -eq 0 ]; then
+  usage $0
+fi
+
+case $1 in
+  start )
+    execute "${DOCKER_COMPOSE_COMMAND} -f compose.yml up -d"
```

Comment:
I am thinking that this may not run from anywhere with this implementation. For instance, can I run adamant_env.sh from a different directory, than this directory, and still have it work? You may need to use the BASH readlink trick to get `$this` to make this work.

Reply:
Added in [243c4cc](https://github.com/lasp/adamant/pull/50/commits/243c4cc778370ac756b75879a5e6f0c304b2890d).